### ISSUE TITLE
Rule Request: AvoidNonPrivateFieldsInSharedObjects

### ISSUE THREAD

**Issue by stokpop** [jborgers/PMD-jPinpoint-rules#117]
Fields in @Component or alike (shared objects) should not contain non-final fields that are not private.

There are no guarantees that the non-private field can be changed by other threads, e.g. in sub class for protected field or other code in case of package private or public field.

Example:

```java
@Controller
public class RootController {

    BasketConfig basketConfig; // not flagged: bad, not private

    @Autowired
    void setBasketConfig(BasketConfig basketConfig) {
        this.basketConfig = basketConfig;
    }
}
```

**Comment by stokpop** [comment]
After talk with Jeroen, it seems better to create a new, separate rule for this:

do not allow non-final non-private fields in shared objects
